src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -31,7 +31,11 @@ class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate
     val recursivePreds = checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)
     val recursivePredIds = recursivePreds.map(_.name)
     val predIdsCalledByRecursivePreds = recursivePreds.flatMap(nonRecursivePredsCalledBy(_, recursivePredIds, input))
-    val cond = { pred: String => !recursivePredIds(pred) && !predIdsCalledByRecursivePreds(pred) }
+    val cond = { pred: String =>
+      input.findPredicate(pred).body.nonEmpty &&
```

Comment:
Don't we do this check up top here?
```Scala
val allPredIds = input.predicates.collect{ case p if p.body.nonEmpty => p.name }.toSet
```
maybe I'm mistaken

Reply:
Yeah, but then I asked you to use `!recursivePredIds` instead of `allPredIds.diff(recursivePredIds)` lmao
`recursivePredIds` won't contain any nonrecursive bodyless predicates

Reply:
lmao we good